attributes.go

```diff
@@ -97,6 +97,13 @@ func addressAttributes(address string) []attribute.KeyValue {
 }
 
 func statusCodeAttribute(protocol string, serverErr error) (attribute.KeyValue, bool) {
+	if connect.IsNotModifiedError(serverErr) {
+		// A "not modified" error is special: it's code is technically "unknown" but
+		// it would be misleading to label it as an unknown error since it's not really
+		// an error, but rather a sentinel to trigger a "304 Not Modified" HTTP status.
+		codeKey := attribute.Key("http.response.status_code")
+		return codeKey.Int(http.StatusNotModified), true
```

Comment:
This seems right directionally. A few thoughts:

* I _think_ we should use [`semconv.HTTPStatusCode`](https://pkg.go.dev/go.opentelemetry.io/otel@v1.16.0/semconv/v1.20.0#HTTPStatusCode) instead of hand-writing a string. (OTel is _really_ larded up with standards, and I think the blessed key for this data is `http.status_code` - there's enough of this nonsense that we should just use the giant package of constants where possible.)
* Can we move this into the Connect-specific portion of the switch statement below? It makes sense to me that we treat this as an actual unknown error for the gRPC protocols, since that's how clients will interpret it. For the Connect protocol, it makes sense to omit the RPC error code and include an HTTP status code instead.

Reply:
Hmm, that is weird that this constant uses `http.status_code`. According to the docs I found, it is `http.response.status_code`: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes

Anyhow, I'll change it to use the constant, and let otel fix either their docs or that constant if one of them is wrong 🤷

Reply:
Done